### PR TITLE
fix: charles review of install

### PR DESCRIPTION
**Pull request by PatrickAlphaC**

### CODE REVIEW

File: gaboon/commands/install.py [hunk 1]
```diff
@@ -166,30 +166,27 @@ def _get_latest_version(org: str, repo: str, headers: dict) -> str:
 def _stream_download(
     download_url: str, target_path: str, headers: dict[str, str] = REQUEST_HEADERS
 ) -> None:
-    response = requests.get(download_url, stream=True, headers=headers)
+    with requests.get(download_url, stream=True, headers=headers) as response:
```
Comment (charles-cooper) [inline gaboon/commands/install.py:169]: not sure this is an improvement

File: gaboon/commands/install.py [hunk 2]
```diff
@@ -166,30 +166,27 @@ def _get_latest_version(org: str, repo: str, headers: dict) -> str:
...
-        raise ConnectionError(
-            f"Received status code {response.status_code} when attempting "
-            f"to download from {download_url}"
-        )
+        temp_file = os.path.join(target_path, "temp_download.zip")
 
-    total_size = int(response.headers.get("content-length", 0))
-    progress_bar = tqdm(total=total_size, unit="iB", unit_scale=True)
-    content = bytes()
+        with open(temp_file, "wb") as f, tqdm(
```
Comment (charles-cooper) [inline gaboon/commands/install.py:175]: or this

File: gaboon/commands/install.py [hunk 3]
```diff
@@ -269,33 +260,64 @@ def _write_dependencies(new_package_ids: list[str], dependency_type: DependencyT
...
+    org: str
+    repo: str
+    version: str | None = None
+
+    @classmethod
+    def from_string(cls, dep_string: str) -> "GitHubDependency":
+        path, version = (
+            dep_string.split("@", 1) if "@" in dep_string else (dep_string, None)
+        )
+        org, repo = str(path).split("/")
```
Comment (charles-cooper) [inline gaboon/commands/install.py:317]: this ternary is a bit hard to read, i suggest
```suggestion
        if "@" in dep_string:
            path, version = dep_string.split("@")
        else:
            path, version = dep_string, None

        org, repo = str(path).split("/")
```

File: gaboon/config.py [hunk 4]
```diff
@@ -215,21 +226,24 @@ def write_dependencies(self, new_dependencies: list):
...
 
     def get_base_dependencies_install_path(self) -> Path:
         project_root = self._project_root
         base_install_path = project_root / self.project.get(
             DEPENDENCIES_FOLDER, DEPENDENCIES_FOLDER
         )
-        base_install_path.mkdir(exist_ok=True)
+        base_install_path.mkdir(
+            exist_ok=True,
+            parents=True,
```
Comment (charles-cooper) [inline gaboon/config.py:245]: weird lint, is `--skip-magic-trailing-comma` enabled?

File: gaboon/commands/install.py [hunk 5]
```diff
@@ -269,33 +260,64 @@ def _write_dependencies(new_package_ids: list[str], dependency_type: DependencyT
...
+
+    if len(new_deps) > 0:
+        config.write_dependencies(new_deps)
+
+
+@dataclass
+class GitHubDependency:
+    org: str
+    repo: str
+    version: str | None = None
```
Comment (charles-cooper) [inline gaboon/commands/install.py:310]: i still find `Optional[str]` clearer, doesn't matter much tho

File: gaboon/commands/install.py [hunk 6]
```diff
@@ -269,33 +260,64 @@ def _write_dependencies(new_package_ids: list[str], dependency_type: DependencyT
...
-                package_org, package_repo = str(package_path).split("/")
-
-                dep_path, _ = dep.split("@", 1) if "@" in dep else (dep, None)
-                dep_org, dep_repo = str(dep_path).split("/")
-                if dep_org == package_org and dep_repo == package_repo:
+                dep_gh = GitHubDependency.from_string(dep)
+                if dep_gh.org == package_dep.org and dep_gh.repo == package_dep.repo:
                     to_delete.add(dep)
-            if package not in to_delete:
+                    updated_packages.add(f"{package_dep.org}/{package_dep.repo}")
```
Comment (charles-cooper) [inline gaboon/commands/install.py:286]: maybe a method for this -- `package_dep.format_no_version()`

File: gaboon/commands/install.py (hunk 1 above)
Comment (PatrickAlphaC) [reply gaboon/commands/install.py:169]: agreed

File: gaboon/config.py (hunk 4 above)
Comment (PatrickAlphaC) [reply gaboon/config.py:245]: https://docs.astral.sh/ruff/settings/#format_skip-magic-trailing-comma

It's defaulted to true in `ruff`. I will change. Good shout.

**Pull request merged**